core/src/Kokkos_View.hpp

```diff
@@ -447,8 +447,13 @@ struct ViewTraits {
  * }
  * \endcode
  */
-template <class DataType, class... Properties>
-class View;
+
+}  // namespace Kokkos
+
+// this is here because it contains the forward declar of Kokkos::View
+#include <impl/Kokkos_ViewTracker.hpp>
```

Comment:
I'd prefer not to put includes in the middle of files. A forward declaration like this should probably go in `Kokkos_Core_fwd.hpp` anyway. Also, when we move a forward declaration that has documentation attached to it, we probably should move the documentation along with it